doc/migrate-to-single-buffer.md

```diff
@@ -0,0 +1,113 @@
+   Suggest to turned off during the upgrading.
+
+#### Detailed steps
+
+1. Create a set of new objects in `BUFFER_POOL`:
+    - `ingress_lossless_pool_new` for both lossless and lossy pools:
+      - copied from `ingress_lossless_pool` with `size` updated to the new size
+    - `egress_lossy_pool_new` with `size` updated:
+      - copied from `egress_lossy_pool` with `size` updated to the new size
+2. Create a set of profiles in `BUFFER_PROFILE` table for ingress lossless/lossy and egress lossy profile with the buffer pools to the ones created in step 1:
+    - `ingress_lossless_profile_new`, `ingress_lossy_profile_new` and `pg_lossless_<speed>_<cable-length>_profile_new`
+      - copied from the profiles without `_new` suffix, with `pool` pointing to `ingress_lossless_pool_new`
+      - `xoff` and `size` updated accordingly for the profiles for lossless headroom
+    - `egress_lossy_profile_new`, `q_lossy_profile_new`: copied from the profiles without `_new` suffix with `pool` pointing to `egress_lossy_pool_new`
+3. Update the BUFFER_PG and BUFFER_QUEUE, making them reference the buffer profiles created in step 2
+4. Update the BUFFER_PORT_INGRESS_PROFILE_LIST and BUFFER_PORT_EGRESS_PROFILE_LIST with the referenced profile replaced
+5. Now we are safe to remove the old pools, profiles. Remove them.
+6. Copy the buffer pools created in step 1 to their original names.
+7. Copy the buffer profiles created in step 2 to their original names
+8. Update the BUFFER_PG and BUFFER_QUEUE with the profiles replaced with the ones created in step 7
+9. Update the BUFFER_PORT_INGRESS_PROFILE_LIST and BUFFER_PORT_EGRESS_PROFILE_LIST
+10. Remove the immediate objects.
+
+### Option 2
+
+#### Resolve the limitation
+
+The root cause of the issue is the orchagent tries modifying the SAI attributes which are create-only. So the solution is straightforward: for `SET` command, check whether the object has already been created:
+
+- if yes, skip the create-only attributes.
+- else, pass all the attributes to SAI.
+
+#### Update the objects in a specific order
+
+Limitations:
+
+1. At any time, for any pool, only one profile using that pool can be in the `BUFFER_PORT_INGRESS_PROFILE_LIST`.
+2. The `pool` attribute of a profile can not be changed. We need to create a temporary pool for lossy traffic to update its pool.
+3. The total pool size can't exceed the maximum accumulative memory.
+
+So the steps should be like this:
+
+1. Copy `ingress_lossy_profile` to a new one named `ingress_lossy_profile_temp` with pool updated to `ingress_lossless_pool`.
+2. Update `BUFFER_PG|<port>|0` with the `profile` updated to `ingress_lossy_profile_temp`.
+3. Update `BUFFER_PORT_INGRESS_PROFILE_LIST` with `ingress_lossy_profile` removed.
+4. Remove `ingress_lossy_profile` and recreate it by duplicating it from `ingress_lossy_profile_temp`.
```

Comment:
Shouldn't the ingress lossy pool get deleted before enlarging the size of the other pools so that we don't exceed the max accumulative memory

Reply:
We don’t need to because
- all the buffer pools are something like threshold, which means to enlarge a pool will succeed if the accumulative size of all pools is greater than the maximum available memory 
- at the time there isn’t any profile referencing the ingress lossy pool, which means it won’t consume any memory, so to remain it for now is harmless

Meanwhile, we can remove the ingress lossy pool here, but the idea is to do it in a “lazy” mode, which means to defer the operations until they’re necessary for the purpose of reducing potential risk.